gammapy/datasets/utils.py

```diff
@@ -67,3 +70,55 @@ def get_nearest_valid_exposure_position(exposure, position=None):
     if not position:
         position = mask_exposure.geom.center_skydir
     return mask_exposure.mask_nearest_position(position)
+
+
+def split_dataset(dataset, width, margin, split_templates=False):
+    geom = dataset.counts.geom.to_image()
+    pixel_width = np.ceil((width / geom.pixel_scales).to_value("")).astype(int)
+
+    ilon = range(0, geom.data_shape[1], pixel_width[1])
+    ilat = range(0, geom.data_shape[0], pixel_width[0])
+
+    datasets = Datasets()
+    for il in ilon:
+        for ib in ilat:
+            l, b = geom.pix_to_coord((il, ib))
+            cutout_kwargs = dict(
+                position=SkyCoord(l, b, frame=geom.frame), width=width + 2 * margin
```

Comment:
if margin is not set, this crashes. Is it wished?

Reply:
yes the margin should be defined so the sources outside the roi that contributes inside are well defined.